packages/entity-server/src/routes/hierarchy/middleware/attachEntityContext.ts

```diff
@@ -73,25 +73,30 @@ const validateEntitiesAndBuildContext = async (
 };
 
 const getFilterInfo = async (
-  req: Request<{ hierarchyName: string }, any, any, { filter?: string }>,
+  req: Request<{ hierarchyName: string }, any, any, { filter?: string; isPublic?: string }>,
   rootEntity: EntityType,
 ) => {
-  const { permission_groups: projectPermissionGroups } = await req.models.project.findOne({
-    code: req.params.hierarchyName,
-  });
+  const isPublic = req.query.isPublic?.toLowerCase() === 'true';
```

Comment:
@EMcQ-BES can I get you to check this side of things, to make sure I don't miss anything crucial in my review?